nntrainer/tensor/tensor.cpp

```diff
@@ -870,6 +870,24 @@ Tensor Tensor::transpose(const std::string &direction) const {
   return result;
 }
 
+Tensor Tensor::dropout_mask(float dropout) const {
+  Tensor result(dim);
+  result.setValue(1.0);
+  Tensor rand_temp(dim);
+  rand_temp.setRandUniform(0.0, 1.0);
+  float scale = 1.0 / (1 - dropout);
+
+  float *mask = result.getData();
+  float *random = rand_temp.getData();
+  for (unsigned int i = 0; i < length(); ++i) {
+    if (random[i] >= dropout)
```

Comment:
please create two loops - one which clips and value to 0/1 and the other which calls element-wise multiplication operation.

Reply:
I cannot see why. could you elaborate?

Reply:
the proposed style reuses blas operations and will be faster, and easier to read as well.

Reply:
But I guess it is much better just use one loop, rather than two.

Reply:
> But I guess it is much better just use one loop, rather than two.

If it is one loop with `if` statements inside it, then I am not sure how well the compiler can optimize.

Reply:
> If it is one loop with `if` statements inside it, then I am not sure how well the compiler can optimize. But this is for optimization part, so lets do it later.

I will fix in later pr.